source/Xunit.Gherkin.Quick/ScenarioXunitHook/ScenarioOutline/ScenarioOutlineDiscoverer.cs

```diff
@@ -20,30 +20,36 @@ public ScenarioOutlineDiscoverer(IMessageSink messageSink)
             ITestMethod testMethod,
             IAttributeInfo factAttribute)
         {
-            var feature = new FeatureDiscoveryModel(new FeatureFileRepository("*.feature")).Discover(testMethod.TestClass.Class.ToRuntimeType());
+            var featureClass = testMethod.TestClass.Class.ToRuntimeType();
+            var featureClassInfo = FeatureClassInfo.FromFeatureClassType(featureClass);
 
-            foreach (var scenarioOutline in feature.Children.OfType<ScenarioOutline>())
+            var features = new FeatureDiscoveryModel(new FeatureFileRepository(featureClassInfo.FileNameSearchPattern)).Discover(featureClass);
+
+            foreach (var feature in features)
             {
-                foreach (var example in scenarioOutline.Examples)
+                foreach (var scenarioOutline in feature.Item2.Children.OfType<ScenarioOutline>())
                 {
-                    var rowIndex = 0;
-                    foreach (var row in example.TableBody)
+                    foreach (var example in scenarioOutline.Examples)
                     {
-                        var tags = feature.GetExamplesTags(scenarioOutline.Name, example.Name);
-                        var skip = feature.IsExamplesIgnored(scenarioOutline.Name, example.Name);
+                        var rowIndex = 0;
+                        foreach (var row in example.TableBody)
+                        {
+                            var tags = feature.Item2.GetExamplesTags(scenarioOutline.Name, example.Name);
+                            var skip = feature.Item2.IsExamplesIgnored(scenarioOutline.Name, example.Name);
```

Comment:
What is this Item2 and item1? Is it possible to avoid using them?

Reply:
Yes, it's possible to avoid them by an ad hoc type. Same as above, it's because of .net tuples.